File: DataStorage/HomeCloud.DataStorage.DataAccess.Contracts/File.cs
namespace HomeCloud.DataStorage.DataAccess.Contracts
{
	#region Usings

	using System;

	#endregion

	/// <summary>
	/// Represents <see cref="File"/> data contract.
	/// </summary>
	public class File
	{
		/// <summary>
		/// Gets or sets the identifier.
		/// </summary>
		/// <value>
		/// The identifier.
		/// </value>
		public Guid ID { get; set; }

		/// <summary>
		/// Gets or sets the directory identifier.
		/// </summary>
		/// <value>
		/// The directory identifier.
		/// </value>
		public Guid DirectoryID {get;set;}

		/// <summary>
		/// Gets or sets the name.
		/// </summary>
		/// <value>
		/// The name.
		/// </value>
		public string Name { get; set; }

		/// <summary>
		/// Gets or sets the extension.
		/// </summary>
		/// <value>
		/// The extension.
		/// </value>
		public string Extension { get; set; }

		/// <summary>
		/// Gets or sets the creation date.
		/// </summary>
		/// <value>
		/// The creation date.
		/// </value>
		public DateTime CreationDate { get; set; }

		/// <summary>
		/// Gets or sets the updated date.
		/// </summary>
		/// <value>
		/// The updated date.
		/// </value>
		public DateTime UpdatedDate { get; set; }
	}
}

File: Management/HomeCloud.Management.Web/ClientApp/app/models/file.ts
export class File {
	private id: string = "";
	private name: string = "";
	private size: number = 0;
	private creationDate: Date = new Date();
	private mimeType: string = "";

	get ID(): string {
		return this.id;
	}
	set ID(value: string) {
		this.id = value;
	}

	get Name(): string {
		return this.name;
	}
	set Name(value: string) {
		this.name = value;
	}

	get CreationDate(): Date {
		return this.creationDate;
	}
	set CreationDate(value: Date) {
		this.creationDate = value;
	}

	get Size(): number {
		return this.size;
	}
	set Size(value: number) {
		this.size = value;
	}

	get MimeType(): string {
		return this.mimeType;
	}
	set MimeType(value: string) {
		this.mimeType = value;
	}
}
